module_test.go

```diff
@@ -0,0 +1,156 @@
+package tls
+
+import (
+	"fmt"
+	"net"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/js/modulestest"
+	"go.k6.io/k6/lib"
+	"go.k6.io/k6/lib/netext"
+	"go.k6.io/k6/lib/types"
+	"go.k6.io/k6/metrics"
+)
+
+func TestGetCertificateOK(t *testing.T) {
+	t.Parallel()
+	trt := newTestRuntime(t)
+
+	ts := httptest.NewTLSServer(nil)
+	defer ts.Close()
+
+	testcases := []string{
+		fmt.Sprintf(`await tls.getCertificate("%s")`, strings.TrimPrefix(ts.URL, "https://")),
+	}
+
+	for i, tc := range testcases {
+		t.Run("testcase#"+strconv.Itoa(i), func(t *testing.T) {
+			_, err := trt.RunOnEventLoop(wrapInAsyncLambda(tc))
+			assert.NoError(t, err)
```

Comment:
Is it worth asserting on the content of the cert here?

Reply:
Done